hook.sh

```diff
@@ -46,17 +46,40 @@ echo -e "\e[32mSetting challenge to ${CERTBOT_VALIDATION} ...\e[0m"
 domain=.$CERTBOT_DOMAIN
 infix=${domain%.$DEDYN_NAME}
 
+# Remove leading wildcard from infix, if present
+# *.foobar.dedyn.io gives "" while *.a.foobar.dedyn.io gives ".a"
+infix=${infix#.\*}
```

Comment:
Cool. Good catch!

Reply:
Am I missing something? If CERTBOT_DOMAIN is "*.a.foobar.dedyn.io" and DEDYN_NAME is foobar.dedyn.io, the domain declaration adds a leading dot, giving ".*.a.foobar.dedyn.io", the first infix declaration gives ".*.a", and the second gives ".a". Or without the wildcard, domain would be .a.foobar.dedyn.io, infix(1) would be ".a", and infix(2) would be ".a". Which we then feed as a suffix to "_acme-challenge", giving "_acme-challenge.a". Or am I missing something?

Reply:
Ah. Of course, that's right. I had overlooked the fact that we add a dot in the beginning. Sorry!